connectors/src/connectors/notion/temporal/activities.ts

```diff
@@ -1550,42 +1545,25 @@ export async function renderAndUpsertPageFromCache({
     });
   }
 
-  localLogger.info("notionRenderAndUpsertPageFromCache: Rendering page.");
-  let renderedPage = "";
   const parsedProperties = parsePageProperties(
     JSON.parse(pageCacheEntry.pagePropertiesText) as PageObjectProperties
   );
+  const renderedPageSection = renderPageSection({
+    blocksByParentId,
+  });
+
   for (const p of parsedProperties) {
     if (!p.text) continue;
     // We skip the title as it is added separately as prefix to the top-level document section.
     if (p.key === "title") continue;
-    renderedPage += `$${p.key}: ${p.text}\n`;
-  }
-  renderedPage += "\n";
-
-  let pageHasBody = false;
-
-  const renderBlock = (b: NotionConnectorBlockCacheEntry, indent = "") => {
-    if (b.blockText) {
-      pageHasBody = true;
-    }
-    let renderedBlock = b.blockText ? `${indent}${b.blockText}\n` : "";
-    const children = blocksByParentId[b.notionBlockId] ?? [];
-    children.sort((a, b) => a.indexInParent - b.indexInParent);
-    for (const child of children) {
-      renderedBlock += renderBlock(child, `- ${indent}`);
-    }
-    return `${renderedBlock}\n`;
-  };
-
-  for (const block of topLevelBlocks) {
-    renderedPage += renderBlock(block);
+    renderedPageSection.sections.push({
```

Comment:
Did not put much thought into that since it will change with the renderSectionContentTitle refactor by @spolu

Reply:
Let's rebase this on https://github.com/dust-tt/dust/pull/3151 as soon as merged.

Aren't you adding them at the end of the document here?

I think we can either add them to the top level content or even have them as prefix for the whole document.... hard to say for sure.

Reply:
> Aren't you adding them at the end of the document here?
yep, just kept that here as reminder

IMHO will add them as a second prefix in addition to title, along with author & lastEditor, using renderPrefix

Reply:
I think we don't want renderPrefix for these do we? this will inevitably be quite messy if we split in the middle of a somewhat structured set of data.

Maybe add a limited number as prefix and the rest as content beyond a count

Reply:
Or just take 3 and overflow if you have more?